### PR TITLE
Add kubectl edit, annotate, autoscale concept to deployment document

### PR DESCRIPTION
**Pull request by janetkuo**
Addresses #15725
The first commit is #16401, please ignore it while reviewing. 

cc @kubernetes/kubectl

### CODE REVIEW

**Comment by bgrant0607** [comment]
Thanks. I'll need to take a look after 1.1 is cut, but we should be able to cherrypick it and push it to gh-pages even after we cut the release candidate.

apply documentation is also in flight. #16401

File: docs/user-guide/managing-deployments.md [hunk 1]
```diff
@@ -411,33 +411,72 @@ You can also run the [update demo](update-demo/) to see a visual representation
...
+
+### kubectl edit
+
+Alternatively, you may also update resources with `kubectl edit`:
+
+```console
+$ kubectl edit pod my-nginx-1jgkf
+```
+
+This is equivalent to first `get` the resource, edit it in text editor, and then `replace` the resource with the updated version:
```
Comment (kargakis) [inline docs/user-guide/managing-deployments.md]: Specify that they can choose an editor by using an env (EDITOR, KUBE_EDITOR, GIT_EDITOR - tbh I would prefer if we just provided EDITOR).

Comment (janetkuo) [inline docs/user-guide/managing-deployments.md]: Added below

Comment (kargakis) [inline docs/user-guide/managing-deployments.md]: Sorry, I didn't mean that we should specify only `EDITOR` here, I was talking in general:) Since we support all three of them, go ahead and add them. Thanks!

Comment (janetkuo) [inline docs/user-guide/managing-deployments.md]: Oh, got it. Updated now.

**Comment by janetkuo** [comment]
@k8s-bot test this

**Comment by janetkuo** [comment]
@k8s-bot e2e test this

**Comment by kargakis** [comment]
LGTM

File: docs/user-guide/managing-deployments.md [hunk 2]
```diff
@@ -411,33 +414,76 @@ You can also run the [update demo](update-demo/) to see a visual representation
...
 
-Sometimes it’s necessary to make narrow, non-disruptive updates to resources you’ve created. For instance, you might want to add an [annotation](annotations.md) with a description of your object. That’s easiest to do with `kubectl patch`:
+Sometimes it’s necessary to make narrow, non-disruptive updates to resources you’ve created. For instance, you might want to update the container's image of your pod.
+
+### kubectl patch
+
+One way to do that is with `kubectl patch`:
 
 ```console
-$ kubectl patch rc my-nginx-v4 -p '{"metadata": {"annotations": {"description": "my frontend running nginx"}}}'
```
Comment (bgrant0607) [inline docs/user-guide/managing-deployments.md]: Could you add a section on annotate?

File: docs/user-guide/managing-deployments.md [hunk 3]
```diff
@@ -411,33 +414,76 @@ You can also run the [update demo](update-demo/) to see a visual representation
...
+### kubectl patch
+
+One way to do that is with `kubectl patch`:
 
 ```console
-$ kubectl patch rc my-nginx-v4 -p '{"metadata": {"annotations": {"description": "my frontend running nginx"}}}' 
-my-nginx-v4
-$ kubectl get rc my-nginx-v4 -o yaml
+# use container name "nginx" as a key to update the image from "nginx" to "busybox"
+$ kubectl patch pod my-nginx-1jgkf -p '{"spec":{"containers":[{"name":"nginx","image":"busybox"}]}}'
```
Comment (bgrant0607) [inline docs/user-guide/managing-deployments.md]: Fixing a typo (e.g., nignx to nginx) would be a more realistic example.

File: docs/user-guide/managing-deployments.md [hunk 4]
```diff
@@ -411,33 +414,76 @@ You can also run the [update demo](update-demo/) to see a visual representation
...
+```console
+$ kubectl apply -f ./nginx-rc.yaml
+replicationcontroller "my-nginx-v4" configured
+```
+
+As shown in the example above, the configuration used with `kubectl apply` is the same as the one used with `kubectl replace`. However, instead of deleting the existing resource and replacing it with a new one, `kubectl apply` modifies the configuration of the existing resource.
+
+Note that `kubectl apply` attaches an annotation to the resource in order to determine the changes to the configuration since the previous invocation. When it's invoked, `kubectl apply` does a three-way diff between the previous configuration, the provided input and the current configuration of the resource, in order to determine how to modify the resource.
+
+Currently, resources are created without this annotation, so the first invocation of `kubectl apply` will fall back to a two-way diff between the provided input and the current configuration of the resource. During this first invocation, it cannot detect the deletion of properties set when the resource was created. For this reason, it will not remove them.
```
Comment (bgrant0607) [inline docs/user-guide/managing-deployments.md]: We should fix apply and update this.

Comment (janetkuo) [inline docs/user-guide/managing-deployments.md]: cc @jackgr

File: docs/user-guide/managing-deployments.md [hunk 5]
```diff
@@ -261,13 +260,32 @@ my-nginx-v4-sh6m8   1/1       Running   0          19m       fe
 my-nginx-v4-wfof4   1/1       Running   0          16m       fe
 ```
 
+For more information, please see [labels](labels.md) and [kubectl label](kubectl/kubectl_label.md) document.
+
+## Updating annotations
+
+Sometimes you want to attach annotations to resources. Annotatations are arbitrary non-identfying metadata for retrieveal by API clients such as tools, libraries, etc. This can be done with `kubectl annotate`. For example:
```
Comment (derekwaynecarr) [inline docs/user-guide/managing-deployments.md]: Typos:  
Annotatations -> Annotations
retrieveal -> retrieval

Comment (janetkuo) [inline docs/user-guide/managing-deployments.md]: Thanks. Fixed.

File: docs/user-guide/managing-deployments.md [hunk 6]
```diff
@@ -275,6 +293,22 @@ my-nginx-divi2   1/1       Running   0          1h
 my-nginx-o0ef1   1/1       Running   0          1h
 ```
 
+To have the system automatically choose the number of nginx replicas as needed, range from 1 to 3, do:
+
+```console
```
Comment (bgrant0607) [inline docs/user-guide/managing-deployments.md]: Was this change intentional?

File: docs/user-guide/managing-deployments.md [hunk 7]
```diff
@@ -387,57 +414,105 @@ and roll it out:
...
-At beginning of loop: my-nginx replicas: 0, my-nginx-v4 replicas: 5
-Updating my-nginx replicas: 0, my-nginx-v4 replicas: 5
-At end of loop: my-nginx replicas: 0, my-nginx-v4 replicas: 5
-Update succeeded. Deleting my-nginx
-my-nginx-v4
+Created my-nginx-v4
+Scaling up my-nginx-v4 from 0 to 5, scaling down my-nginx from 4 to 0 (keep 4 pods available, don't exceed 5 pods)
+Scaling my-nginx-v4 up to 1
+Scaling my-nginx down to 3
+Scaling my-nginx-v4 up to 2
```
Comment (bgrant0607) [inline docs/user-guide/managing-deployments.md]: Thanks for updating this.

File: docs/user-guide/managing-deployments.md (hunk 6 above)
Comment (janetkuo) [inline docs/user-guide/managing-deployments.md]: Yes. It's an appropriate place to introduce `autoscale` imo.

**Comment by bgrant0607** [comment]
Looks pretty good. We should submit Jack's PR and then rebase this one.

**Comment by bgrant0607** [comment]
cc @jackgr

File: docs/user-guide/managing-deployments.md [hunk 8]
```diff
@@ -275,6 +293,22 @@ my-nginx-divi2   1/1       Running   0          1h
...
+$ kubectl get horizontalpodautoscaler 
+NAME      REFERENCE                           TARGET    CURRENT     MINPODS   MAXPODS   AGE
+nginx     ReplicationController/nginx/scale   80%       <waiting>   1         3         1m
+```
+
+For more information, please see [kubectl scale](kubectl/kubectl_scale.md), [kubectl autoscale](kubectl/kubectl_autoscale.md) and [horizontal pod autoscaler](horizontal-pod-autoscaling/README.md) document.
+
 ## Updating your application without a service outage
 
 At some point, you’ll eventually need to update your deployed application, typically by specifying a new image or image tag, as in the canary deployment scenario above. `kubectl` supports several update operations, each of which is applicable to different scenarios.
```
Comment (kargakis) [inline docs/user-guide/managing-deployments.md]: This makes me think if we want to start providing `rolling-update` options like choosing scaling direction, max surge, and max unavailable @ironcladlou @smarterclayton @bgrant0607

**Comment by kargakis** [comment]
still lgtm

File: docs/user-guide/managing-deployments.md (hunk 8 above)
Comment (bgrant0607) [inline docs/user-guide/managing-deployments.md]: I agree, @kargakis. I do hope to steer most users to Deployment, but we might also want a rolling-update interface to Deployment.
cc @nikhiljindal

**Comment by janetkuo** [comment]
Squashed commits; wait for #16401.

File: docs/user-guide/managing-deployments.md [hunk 9]
```diff
@@ -387,57 +414,105 @@ and roll it out:
...
+# do some edit, and then save the file
 $ kubectl replace -f /tmp/nginx.yaml
-replicationcontrollers/my-nginx-v4
-$ rm $TMP
+pod "my-nginx-1jgkf" replaced
+$ rm /tmp/nginx.yaml
 ```
 
-The system ensures that you don’t clobber changes made by other users or components by confirming that the `resourceVersion` doesn’t differ from the version you edited. If you want to update regardless of other changes, remove the `resourceVersion` field when you edit the resource. However, if you do this, don’t use your original configuration file as the source since additional fields most likely were set in the live state.
+This allows you to do more significant changes more easily. Note that you can specify the editor with your `EDITOR`, `KUBE_EDITOR`, or `GIT_EDITOR` environment variables.
```
Comment (janetkuo) [inline docs/user-guide/managing-deployments.md]: Remove GIT_EDITOR; ref #17108

File: docs/user-guide/managing-deployments.md (hunk 1 above)
Comment (janetkuo) [inline docs/user-guide/managing-deployments.md]: Removed GIT_EDITOR; ref #17108

**Comment by kargakis** [comment]
somebody tag this lgtm
